libqtile/lazy.py

```diff
@@ -90,18 +103,23 @@ def kwargs(self) -> Dict:
         """The kwargs to the given call"""
         return self._kwargs
 
-    def when(self, layout: Optional[Union[Iterable[str], str]] = None,
+    def when(self, focused: Optional['Match'] = None, layout: Optional[Union[Iterable[str], str]] = None,
```

Comment:
IIRC the preferred way for Python 3.10+ is to state the objects themselves directly in type annotations, i.e. `Optional[Match]` with the conditional import. In previous versions this would fail with a `NameError` but from py310 type annotation evaluation is delayed, so the exception is not raised during runtime. When type checking the objects are imported and it works. Before py310 the exception is avoided by using `from __future__ import annotations` which many of our modules have too. You may need to add that to the top of this file too to avoid the exception in the runtimes for py <3.10.

Reply:
Ah makes sense, thanks for the detailed info.

Reply:
I used `from __future__ import annotations` which worked perfectly. Thanks again